common/stream.h

```diff
@@ -304,21 +319,34 @@ inline void stream_in_t::pop(char* buffer, uint64_t bufferSize)
 		return;
 	}
 
-	memcpy(buffer, &this->inBuffer[inPosition], bufferSize);
+	if (bufferSize == 0)
+	{
+		return;
+	}
 
+	std::copy(&this->inBuffer[inPosition], &this->inBuffer[inPosition] + bufferSize, buffer);
```

Comment:
why `this->inBuffer`? just `inBuffer` is enough.
Personally, I would use iterators with `std::copy`:
```cpp
    auto bufferStart = inBuffer.begin() + inPosition;

    std::copy(bufferStart, bufferStart + bufferSize, buffer);
```